Close #172 #158 #165 Doc update for PR template, description, and R Version in programming strategy (#173)
pharmaverse/admiraldev: .github/pull_request_template.md

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files

> - [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)

Reviewer: The Unit Test guide doesn't mention any "edge cases", what exactly is an edge case? and should it be included in here?

Author: Maybe I should use https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios ?

Reviewer: I think the entire document is relevant so I'm not sure I would point to a specific section of the unit test guide, but are there any documentations that provide further details about edge cases, if not I think pointing to the unit guide might be better..

@bms63 thoughts?

Reviewer 2: I don't think we want to get to in the weeds with edge cases in the PR Template, but we could expand on them in the guidance.   For example, edge case might be NAs, partial data, points almost at the <= or >= cut-offs, weird characters that can pop up in data we deal with.  Perhaps a few examples from the test files would suffice??

I didn't realize we had a section on realistic data scenarios.

Reviewer 2: I think we wanted to prompt people to think about edge cases in teh PR templates - as this has been some incidents where we could of caught bugs earilier if we had been more thoughtful in our edge cases.

Author: Looks like I may have pulled the trigger a little too quick there for another commit. I also thought it was checkbox with the link was more as a reminder and once you comb over the unit-test doc a couple times it feels clear to try and come up with "weird" test cases

Author: Should I just add an additional blurb on the guidance for this PR as well or make it a separate issue? @bms63

> - [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)

Reviewer 2: ```suggestion
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
```

Reviewer 2: what about this?

Author: Make sense to me to match previously written "lingo"

Reviewer: I think that sounds good!

> - [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1)?

Reviewer: is the `-1` here needed ?`https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1`

> 

Reviewer 2: new line?

Author: I had copy/pasted the exact same template from admiral, I realize now they were actually different?

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files

> - [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)

Reviewer 2: ```suggestion
- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
```

Reviewer 2: actually does this link work?

> - [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?

Reviewer 2: ```suggestion
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1)?
```

Reviewer 2: can you make sure all these links work please

Author: Links indeed work, small caveat is that /devel/ website can go back to /main/ website, but not the other way around, but I think this is a separate issue

Reviewer 2: yes i think that has to do with the `readme.Rmd` file.  I think there is an issue to investigate

- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)

> - [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 

Reviewer 2: I guess since this admiraldev we should point to the Reference for that package down
